lib/chess_web/controllers/registration_controller.ex

```diff
@@ -12,8 +13,9 @@ defmodule ChessWeb.RegistrationController do
     changeset = User.changeset(%User{}, user)
 
     case Repo.insert(changeset) do
-      {:ok, _user} ->
+      {:ok, user} ->
         conn
+        |> Guardian.Plug.sign_in(user)
```

Comment:
Lol @ "_could_ be", Hound :wink: .

So this looks like a learning opportunity for me - why would you _not_ alias to the "leaf node" module? Why wouldn't you alias `Chess.Auth.Guardian.Plug` ? I had thought maybe other modules of `Guardian` might be referenced in this file, but diddn't see any. Is there any other reason besides consistency?

Reply:
Yeah, I think in this case I would leave as `Guardian.Plug` to avoid confusion with `Plug` the library.

Reply:
I think this is very much a personal preference thing. It makes sense to me to alias modules I've defined (`Chess.Auth.Guardian` in this case).

It wasn't specifically to avoid a collision with Elixir's `Plug` but now you mention it that's a nice side effect!